test/org/sagebionetworks/bridge/play/controllers/AuthenticationControllerMockTest.java

```diff
@@ -699,38 +710,115 @@ private void mockResetPasswordRequest() throws Exception {
         assertEquals(TEST_EMAIL, deser.getEmail());
     }
     
+    @Test
+    public void requestResetPasswordWithPhone() throws Exception {
+        mockSignInWithPhonePayload();
+        study.getMinSupportedAppVersions().put(OperatingSystem.IOS, 0);
+        
+        controller.requestResetPassword();
+        
+        verify(authenticationService).requestResetPassword(eq(study), signInCaptor.capture());
+        SignIn deser = signInCaptor.getValue();
+        assertEquals(TEST_STUDY_ID_STRING, deser.getStudyId());
+        assertEquals(TestConstants.PHONE.getNumber(), deser.getPhone().getNumber());
+    }
+    
     @Test(expected = UnsupportedVersionException.class)
     public void requestResetPasswordAppVersionDisabled() throws Exception {
-        mockEmailRequestPayload();
+        mockSignInWithEmailPayload();
         study.getMinSupportedAppVersions().put(OperatingSystem.IOS, 20); // blocked
         
         controller.requestResetPassword();
     }
 
     @Test(expected = EntityNotFoundException.class)
     public void requestResetPasswordNoStudy() throws Exception {
-        TestUtils.mockPlayContextWithJson(new Email((StudyIdentifier) null, TEST_EMAIL));
+        when(studyService.getStudy((String)any())).thenThrow(new EntityNotFoundException(Study.class));
+        TestUtils.mockPlayContextWithJson(new SignIn.Builder().withEmail(TEST_EMAIL).build());
         controller.requestResetPassword();
     }
-
-    private void mockEmailRequestPayload() throws Exception {
-        Map<String,String[]> headers = new ImmutableMap.Builder<String,String[]>()
-                .put("User-Agent", new String[]{"App/14 (Unknown iPhone; iOS/9.0.2) BridgeSDK/4"}).build();
-        String json = TestUtils.createJson(
-                "{'study':'" + TEST_STUDY_ID_STRING +"','email':'"+TEST_EMAIL+"'}");
-        TestUtils.mockPlayContextWithJson(json, headers);
+    
+    @Test
+    public void requestPhoneSignIn() throws Exception {
+        TestUtils.mockPlayContextWithJson(PHONE_SIGN_IN_REQUEST);
+        
+        Result result = controller.requestPhoneSignIn();
+        TestUtils.assertResult(result, 202, "Message sent.");
+        
+        verify(authenticationService).requestPhoneSignIn(signInCaptor.capture());
+        
+        SignIn captured = signInCaptor.getValue();
+        assertEquals(TEST_STUDY_ID_STRING, captured.getStudyId());
+        assertEquals(TestConstants.PHONE.getNumber(), captured.getPhone().getNumber());
+    }
+    
+    @Test
+    public void phoneSignIn() throws Exception {
+        TestUtils.mockPlayContextWithJson(PHONE_SIGN_IN);
+        
+        Metrics metrics = new Metrics(TEST_REQUEST_ID);
+        doReturn(metrics).when(controller).getMetrics();
+        
+        when(authenticationService.phoneSignIn(any(), any())).thenReturn(userSession);
+        
+        Result result = controller.phoneSignIn();
+        assertEquals(200, result.status());
```

Comment:
Is it worth verifying that the result contains a user session?

Reply:
Added